platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/NonAncientEventWindow.java

```diff
@@ -33,8 +33,74 @@ public class NonAncientEventWindow {
 
     private final AncientMode ancientMode;
     private final long latestConsensusRound;
-    private final long minRoundNonAncient;
-    private final long minGenNonAncient;
+
+    private final long ancientThreshold;
+    private final long expiredThreshold;
+
+    /**
+     * Create a NonAncientEventWindow by calculating the minRoundNonAncient value from the latestConsensusRound and
+     * roundsNonAncient.
+     *
+     * @param latestConsensusRound the latest round that has come to consensus
+     * @param minGenNonAncient     the minimum generation that is non-ancient
+     * @param minGenNonExpired     the minimum generation that is non-expired
+     * @param roundsNonAncient     the number of rounds that are non-ancient
+     * @param roundsNonExpired     the number of rounds that are non-expired
+     * @return the new NonAncientEventWindow
+     */
+    @NonNull
+    public static NonAncientEventWindow createUsingRoundsNonAncient(
+            final long latestConsensusRound,
+            final long minGenNonAncient,
+            final long minGenNonExpired,
+            final long roundsNonAncient,
+            final long roundsNonExpired,
+            @NonNull final AncientMode ancientMode) {
```

Comment:
missing javadoc for `ancientMode`

Reply:
I've drastically simplified the constructors for this class. I was getting super confused when trying to fix unit tests having multiple constructors with slightly different lists of arguments, and where most of the arguments are just long values.